rtribbia/question2.js

```diff
@@ -0,0 +1,98 @@
+//---------------------------
+//  2A
+//---------------------------
+// I was following the example at the end of 2/9 scratchpad and, to be honest, it works but I don't really know what's going on in the subclass functions starting at var proto.
+
+var Animal = (function () {
+	function Animal(name) {
+		this.name = name;
+		this.layEgg = function() {
+			return (new Egg(this.__proto__.constructor));
+		};
+	}
+
+	Animal.prototype.move = function () {
+		return 'walk';
+	}
+
+	return Animal;
+})();
+
+var Bird = (function (Super) {
+	function Bird(name) {
+		Super.call(this,name);
+	}
+
+	var proto = new Super([]);
+	Bird.prototype = proto;
+	proto.constructor = Bird;
+	proto.move = function() {
+		return 'fly'
+	}
+	proto.hasWings = true;
+
+	return Bird;
+
+})(Animal);
```

Comment:
The first thing to note here is that you're calling `Bird` and passing in the parameter `Animal` here on line 36. `Animal` is referred to internally as `Super` (on line 21). So when you define `proto`, you're making it a new `Animal`. Then you set the prototype of the `Bird` function to be that Animal you just created, and set that Animal's constructor to be `Bird`. That's the heart of subclassing--taking an instance of the parent class and using it as the blueprint for all instances of the subclass.  Lines 29-32 give new properties to the prototype, which is how you make instances of the subclass different from the parent class.